### PR TITLE
add new-app support for detecting .net apps

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@csrwng ptal

**Comment by bparees** [comment]
@jupierce in the interest of knowledge sharing, you may take a look as well :)

**Comment by csrwng** [comment]
@bparees do we want to also update the example detection script?
https://github.com/openshift/origin/blob/master/examples/gitserver/hooks/detect-language

The purpose of that script was to give you an alternate language detection mechanism that could be custom to your organization. By creating apps through a git push instead of invoking new-app directly, you can fine tune which images are used.

**Comment by bparees** [comment]
@csrwng turns out this actually does work as is because of:
https://github.com/openshift/origin/blob/master/pkg/generate/app/sourcelookup.go#L442-L449

was already appending all the types from each detector that matched.

and the code i ripped out was unused and thus the tests were pointless.

File: examples/gitserver/hooks/detect-language [hunk 1]
```diff
@@ -50,4 +50,9 @@ if has index.pl cpanfile; then
   exit 0
 fi
 
+if has index.php project.json; then
```
Comment (csrwng) [inline examples/gitserver/hooks/detect-language]: index.php?

**Comment by csrwng** [comment]
Just one comment on the detection script, otherwise LGTM

File: examples/gitserver/hooks/detect-language (hunk 1 above)
Comment (bparees) [inline examples/gitserver/hooks/detect-language]: yuck. thanks.

**Comment by bparees** [comment]
[merge]

**Pull request merged**